sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -59,6 +66,8 @@ def __init__(self) -> None:
         self.incoming_edges: MutableMapping[str, list[str]] = defaultdict(list)
         self.outgoing_edges: MutableMapping[str, list[str]] = defaultdict(list)
         self.sources: list[Source] = []
+        self.root: Optional[Step] = None
+        self.__previous_step: Optional[Step] = None
```

Comment:
What is this meant for ?

Reply:
The previous step is used to determine the edge when a step registers. The pipeline passes in the previous step so that edge can be registered correctly.

The root is used mostly when branches are merged, to easily find the first step in the pipeline.